### PR TITLE
Fix/package updates

### PR DESCRIPTION
**Pull request by divyarajmasani**
## Context

Post rebase onto main, `feature/embed-stories` needed to update some of the package usage.

## Summary

This PR updates the packages used to `@web-stories-wp` package. And fixes default options for the customizer.

### CODE REVIEW

File: assets/src/tinymce/components/Modal.js [hunk 1]
```diff
@@ -18,6 +18,8 @@
  * External dependencies
  */
 import PropTypes from 'prop-types';
+import { __ } from '@web-stories-wp/i18n';
```
Comment (swissspidy) [inline assets/src/tinymce/components/Modal.js:21]: Everything that is _not_ story editor or story dashboard should keep using `@wordpress/i18n`.

So need to undo that change here :-)

File: assets/src/web-stories-block/block/block-types/selected-stories/storyPreview.js [hunk 2]
```diff
@@ -17,6 +17,7 @@
  * External dependencies
  */
 import styled from 'styled-components';
+import { getRelativeDisplayDate } from '@web-stories-wp/date';
```
Comment (swissspidy) [inline assets/src/web-stories-block/block/block-types/selected-stories/storyPreview.js:20]: Any chance we can use `@wordpress/date` to display the relative dates here?

Our blocks should ideally _not_ be using `@web-stories-wp/date`

File: includes/Integrations/Core_Themes_Support.php [hunk 3]
```diff
@@ -79,11 +80,12 @@ public function extend_theme_support() {
 	 * @return void
 	 */
 	public function embed_web_stories() {
-		$customizer = new Customizer();
-		$this->enqueue_style( 'web-stories-theme-style-' . get_stylesheet(), [ Renderer::STYLE_HANDLE ] );
+		if ( is_readable( WEBSTORIES_PLUGIN_DIR_PATH . '/assets/css/web-stories-theme-style-' . get_stylesheet() . '.css' ) ) {
```
Comment (swissspidy) [inline includes/Integrations/Core_Themes_Support.php:83]: Why was this added?

File: includes/Model/Story.php [hunk 4]
```diff
@@ -133,8 +133,8 @@ public function load_from_post( $_post ) {
 			return false;
 		}
 
-		$this->title   = get_the_title( $post );
-		$this->excerpt = get_the_excerpt( $post );
+		$this->title   = $post->post_title;
+		$this->excerpt = $post->post_excerpt;
```
Comment (swissspidy) [inline includes/Model/Story.php:137]: Why was this changed?

File: includes/Integrations/Core_Themes_Support.php [hunk 5]
```diff
@@ -79,11 +80,12 @@ public function extend_theme_support() {
...
+			$this->enqueue_style( 'web-stories-theme-style-' . get_stylesheet(), [ Renderer::STYLE_HANDLE ] );
+		}
 		?>
 		<div class="web-stories-theme-header-section">
-			<?php echo $customizer->render_stories(); // phpcs:ignore -- WordPress.Security.EscapeOutput.OutputNotEscaped - Escaped web stories HTML. ?>
+			<?php render_theme_stories(); ?>
```
Comment (swissspidy) [inline includes/Integrations/Core_Themes_Support.php:88]: Neat!

File: assets/src/tinymce/web-stories.js [hunk 6]
```diff
@@ -14,12 +14,17 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import { __ } from '@web-stories-wp/i18n';
```
Comment (swissspidy) [inline assets/src/tinymce/web-stories.js:20]: Keep using `@wordpress/i18n` here

File: includes/Model/Story.php (hunk 4 above)
Comment (divyarajmasani) [reply includes/Model/Story.php:137]: on web-stories archive pages we run few filters specifically `the_excerpt` and `the_content`, in this filters we try to embed the player. 
https://github.com/rtCamp/web-stories-wp/blob/49b01cea256adf5367155bcf84d5f430c4a78bf7/includes/Story_Post_Type.php#L714

Now here we are using `$story->load_from_post( $post );` which in turn used `get_the_excerpt()`. And this technically shouldn't collide. but on the archive page due to this it was going in too deep and resulting in page not loading and causing 502 server error.

Using data already fetched above with `get_post()` resolved the issue.

File: assets/src/tinymce/web-stories.js (hunk 6 above)
Comment (divyarajmasani) [reply assets/src/tinymce/web-stories.js:20]: I had changed this because eslint throw an error for this file, I think we may then update the eslint rule application.

File: includes/Integrations/Core_Themes_Support.php (hunk 3 above)
Comment (divyarajmasani) [reply includes/Integrations/Core_Themes_Support.php:83]: We don't have stylesheets for some themes like `2019`, there it was going 404. so better to check if the file exists.

File: assets/src/tinymce/components/Modal.js (hunk 1 above)
Comment (divyarajmasani) [reply assets/src/tinymce/components/Modal.js:21]: Yep, as mentioned below, this change was warranted by the eslint, will check the eslint rules and see if we need to exclude this files.

File: assets/src/tinymce/web-stories.js (hunk 6 above)
Comment (swissspidy) [reply assets/src/tinymce/web-stories.js:20]: What error did you get specifically? There are a few possibilities.

But yes it's possible that we need to update `.eslintrc`.

File: includes/Model/Story.php (hunk 4 above)
Comment (swissspidy) [reply includes/Model/Story.php:137]: Got it! Bit of a recursion there.

Change makes sense then.

I'll need to think about improving the archive page anyway 🤔

File: assets/src/tinymce/web-stories.js (hunk 6 above)
Comment (divyarajmasani) [reply assets/src/tinymce/web-stories.js:20]: > '@wordpress/i18n' import is restricted from being used. Use @web-stories-wp/i18n instead.(no-restricted-imports)

This one, guess need to update the config to exclude tinymce files from this rule.

File: includes/Model/Story.php (hunk 4 above)
Comment (divyarajmasani) [reply includes/Model/Story.php:137]: we can now make the archive page use grid view. with Lightbox, but if we need individual click through for the single stories then may be disable the lightbox for archive page. 🤔

File: assets/src/tinymce/web-stories.js (hunk 6 above)
Comment (swissspidy) [reply assets/src/tinymce/web-stories.js:20]: Yes. It's at the end of the `.eslintrc` file. Although no idea why it is being picked up by that rule 🤔

File: includes/Model/Story.php (hunk 4 above)
Comment (swissspidy) [reply includes/Model/Story.php:137]: Archive pages are styled by the theme now, so can't just make a grid.

But we have an existing ticket for allowing user customization.

Anyway, not a concern right now :-)

File: assets/src/tinymce/web-stories.js (hunk 6 above)
Comment (divyarajmasani) [reply assets/src/tinymce/web-stories.js:20]: tinyMCE was added to the eslintrc, and I believe during rebase it showed conflict and to resolve it, must have added the file in the `files` array.

**Review by swissspidy — APPROVED** [review]

**Comment by spacedmonkey** [comment]
Can you check the js unit tests / e2e tests. They seem to be fail for valid reasons. 

Also, can you revert changes to package lock file. It is 20k changes seems odd.

**Comment by divyarajmasani** [comment]
> Can you check the js unit tests / e2e tests. They seem to be fail for valid reasons.
> 
> Also, can you revert changes to package lock file. It is 20k changes seems odd.

About the package-lock.json, I had to use `--legacy-peer-deps`  for resolving some package-lock issues. And I guess this is the result of that. 

While doing this npm sometimes removes `resolved` url, or `dev: true` from `package-lock.json`. Post rebase when I ran the `npm i` on this branch it again added those `resoled` url. Not sure if this behaviour is reported anywhere. suggest @spacedmonkey

**Comment by divyarajmasani** [comment]
Updated the PR, removed unrelated package-lock changes. Will work on resolving the test case.

**Comment by divyarajmasani** [comment]
@spacedmonkey failing JS unit test is for the `@web-stories-wp/date` package
 
due to following line, it is return smallcase `am,pm`.
https://github.com/rtCamp/web-stories-wp/blob/1a163370fec08faca262b5ea0c6ade9e947a3ea7/packages/date/src/convertFormatString.js#L117

> https://date-fns.org/v2.17.0/docs/format

For capital `AM,PM` we may have to use `a`. 🤔

**Comment by swissspidy** [comment]
> @spacedmonkey failing JS unit test is for the `@web-stories-wp/date` package
> 
> due to following line, it is return smallcase `am,pm`.
> https://github.com/rtCamp/web-stories-wp/blob/1a163370fec08faca262b5ea0c6ade9e947a3ea7/packages/date/src/convertFormatString.js#L117
> 
> > [date-fns.org/v2.17.0/docs/format](https://date-fns.org/v2.17.0/docs/format)
> 
> For capital `AM,PM` we may have to use `a`. 🤔

You're right. `A` and `a` are incorrect there. Should be:

```
// Lowercase Ante meridiem and Post meridiem. am or pm.
a: 'aaa',

// Uppercase Ante meridiem and Post meridiem. AM or PM
A: 'aa',
```

BUT: why is this failing _now_? What changed to cause this? And why is it not failing in `google/web-stories-wp`?

**Comment by swissspidy** [comment]
> > Can you check the js unit tests / e2e tests. They seem to be fail for valid reasons.
> > Also, can you revert changes to package lock file. It is 20k changes seems odd.
> 
> About the package-lock.json, I had to use `--legacy-peer-deps` for resolving some package-lock issues. And I guess this is the result of that.
> 
> While doing this npm sometimes removes `resolved` url, or `dev: true` from `package-lock.json`. Post rebase when I ran the `npm i` on this branch it again added those `resoled` url. Not sure if this behaviour is reported anywhere. suggest @spacedmonkey
> 
> Also there is this dependency conflict in e2e-test package,
> Screenshot 2021-02-25 at 12 13 26 PM

Let's ignore the lock file changes for now

**Comment by divyarajmasani** [comment]
> BUT: why is this failing _now_? What changed to cause this? And why is it not failing in `google/web-stories-wp`?

I don't think we have changed anything in the packages though testing it on both the branches again and will try to see.

**Comment by swissspidy** [comment]
As discussed, let's try fixing the broken code here (because it's clearly a bug)

**Comment by swissspidy** [comment]
@divyarajmasani do we need to update `a` as well? so:

```
// Lowercase Ante meridiem and Post meridiem. am or pm.
a: 'aaa',

// Uppercase Ante meridiem and Post meridiem. AM or PM
A: 'aa',
````

**Comment by divyarajmasani** [comment]
@swissspidy 
I think that one is fine it's as per the doc here,
> https://date-fns.org/v2.17.0/docs/format

**Comment by swissspidy** [comment]
Maybe I am missing something, but `a` in [PHP date format](https://www.php.net/manual/en/datetime.format.php) prints `am`/`pm`.

To get `am`/`pm` in `date-fns` we need to use `aaa`.

Right now we use `aaaa` though, which would yield `a.m`/`p.m.`, so that's different.

**Comment by divyarajmasani** [comment]
> Maybe I am missing something, but `a` in [PHP date format](https://www.php.net/manual/en/datetime.format.php) prints `am`/`pm`.
> 
> To get `am`/`pm` in `date-fns` we need to use `aaa`.
> 
> Right now we use `aaaa` though, which would yield `a.m`/`p.m.`, so that's different.

you are right there, I missed that you posted a suggestion and not the actual code, yeah for `am / pm` will have to update it to `aaa`, on it. I'll update the PR,

**Comment by divyarajmasani** [comment]
@swissspidy I've updated it to use `aaa` for `am / pm`.

**Review by swissspidy — APPROVED** [review]
